### PR TITLE
use temp file for kubeconfig in test

### PR DESCRIPTION
**Pull request by juanvallejo**
Followup to https://github.com/kubernetes/kubernetes/pull/67093

Updates client_config_test to use a temporary file for kubeconfig.

**Release note**:
```release-note
NONE
```

cc @deads2k @soltysh

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/client-go/tools/clientcmd/client_config_test.go [hunk 1]
```diff
@@ -132,9 +132,19 @@ func TestModifyContext(t *testing.T) {
 		"clean":   true,
 	}
 
+	tempPath, err := ioutil.TempFile("", "testclientcmd-")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	defer func() {
+		os.Remove(tempPath.Name())
```
Comment (neolit123) [inline staging/src/k8s.io/client-go/tools/clientcmd/client_config_test.go]: given this is a single call to be defered can we do?
```
defer os.Remove(tempPath.Name())
```

Comment (juanvallejo) [reply staging/src/k8s.io/client-go/tools/clientcmd/client_config_test.go]: Thanks, updated.

**Pull request merged**